src/test_util/mod.rs

```diff
@@ -348,6 +348,64 @@ where
     .await
 }
 
+// Retries a func every `retry` duration until given an Ok(T); panics after `until` elapses
+pub async fn retry_until<F, Fut, T, E>(mut f: F, retry: Duration, until: Duration) -> T
+where
+    F: FnMut() -> Fut,
+    Fut: Future<Output = Result<T, E>> + Send + 'static,
+{
+    let started = Instant::now();
+    while started.elapsed() < until {
+        match f().await {
+            Ok(res) => return res,
+            Err(_) => tokio::time::delay_for(retry).await,
+        }
+    }
+    panic!("Timeout")
+}
+
+#[cfg(test)]
+mod tests {
+    use super::retry_until;
+    use std::{
+        sync::{Arc, RwLock},
+        time::Duration,
+    };
+
+    #[allow(clippy::never_loop)]
+    async fn retry_until_helper(count: Arc<RwLock<i32>>) -> Result<(), ()> {
+        while *count.read().unwrap() < 3 {
+            let mut c = count.write().unwrap();
+            *c += 1;
+            return Err(());
+        }
+        Ok(())
+    }
```

Comment:
Why while-loop used here? :thinking:

Reply:
Thanks, this is result of some copying and pasting from an earlier inline version that I didn't replace. It's redundant. Will remove.